pkg/kapp/diff/explicit_versioned_ref.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+)
+
+type VersionedRefDesc struct {
+	Namespace string `json:"namespace"`
+	APIGroup  string `json:"apiGroup"`
+	Kind      string `json:"kind"`
+	Name      string `json:"name"`
+}
+
+type ExplicitVersionedRefAnn struct {
+	References []VersionedRefDesc `json:"references"`
+}
+
+type ExplicitVersionedRef struct {
+	Resource   VersionedResource
+	Annotation ExplicitVersionedRefAnn
+}
+
+const (
+	explicitReferenceKey = "kapp.k14s.io/versioned-explicit-ref"
+)
+
+func NewExplicitVersionedRef(res VersionedResource, annotation ExplicitVersionedRefAnn) *ExplicitVersionedRef {
+	return &ExplicitVersionedRef{res, annotation}
+}
+
+func (e *ExplicitVersionedRef) IsReferenced() bool {
+	refList := e.refStringList()
+	resRef := e.resourceRefString()
```

Comment:
Should we have something like?
```suggestion
	references := e.references()
	referenceKey := e.resourceRefKey()
```

Reply:
Yep that's right! That does seem to be more intuitive I'll make that change

Reply:
Resolved!

Reply:
Should we also change the function names?

Reply:
Makes sense to, will do!